### DISCUSSION TITLE
Upgrade to release v3.0.0 fails

### DISCUSSION THREAD

**Discussion by escuco** [JulianPrieber/littlelink-custom#212] [Issues]
I'm not able to perform the upgrade to v3.0.0 via the web backend.
When upgrading from v2.9.2 through the web backend a `404  not found` is thrown out directly after the backup process at this url: `https://my-domain/update?preparing`.

After that, I can access the web backend manually and the upgrade is still available.

All upgrades before this one went fine.

**Comment by JulianPrieber** [comment]
This should be an easy fix.
During the update, the advanced config entry `custom_url_prefix` cannot be empty.
Just enter anything there before the update. After the update, you can empty the entry again.

This error will not appear when update from 3.0.0 to future versions.

Hope this helps 👍

**Comment by root9191** [comment]
My Update also failed and now i only get the Maintenance Message:
```
Maintenance Mode
We are performing scheduled site maintenance at this time.
Please check back with us later.
```
The custom_url_prefix looks like this: 
`'custom_url_prefix' => '+',`
And in the .env File the Maintenance Mode is "false".
Could you help me out here?

**Comment by JulianPrieber** [comment]
In your storage folder, there is a file called "MAINTENANCE". If you delete that, you can get out of the maintenance mode.

**Comment by root9191** [comment]
That worked to get out of the Maintenance Mode, thanks!
The Parameter now looks like this, but the Update still fails:
`'custom_url_prefix' => 'test',`
![Bildschirmfoto 2022-11-15 um 18 20 39](https://user-images.githubusercontent.com/47309435/201986827-f302a804-7ef6-4a8e-868e-7ad4866a6979.png)

**Comment by JulianPrieber** [comment]
That means your page cannot migrate the new database entries.

This can be caused if PHP isn't allowed to run system commands.

There are a few ways of solving this. The easiest would be if you have shell access to your LLC instance.
Can you run system commands in your setup?

**Comment by root9191** [comment]
Ok i understand. Unfortunately i cannot access LLC via shell. 
It is hosted by hosttech and there i can only access via Webbrowser or FTP.

**Comment by JulianPrieber** [comment]
That explains why the update fails. The updater cannot change database entries on pretty much any hosting provider.

The easiest way would be to upload the database from the manual update version, but this would effectively reset your instance.

Otherwise, you'd have to find a way to add the new database entries manually.

**Comment by root9191** [comment]
What if i download the Database and add it locally and then upload it again?

**Comment by JulianPrieber** [comment]
You could download your database, put it in a local instance and run the database seeder of the updater (update?finishing) then upload it again.

**Comment by escuco** [comment]
Hello!

> This should be an easy fix. During the update, the advanced config entry `custom_url_prefix` cannot be empty. Just enter anything there before the update. After the update, you can empty the entry again.

That did the trick! Thank you so much.

Cheers!
